### ISSUE TITLE
GitHub IDP brings unwanted dependencies even if not used - Retrofit and Retrofit GSON

### ISSUE THREAD

**Issue by curiousily** [firebase/FirebaseUI-Android#1412]
The (somewhat) recently introduced GitHub IDP adds dependencies to Retrofit and Retrofit GSON. I am not aware of the number of users of Firebase Auth that will use this exact IDP but bringing those libraries as required seems strange. Our app does not use any of these libraries.

Please, make the GitHub IDP optional or make optional including the Retrofit, Retrofit Gson and GSON (not sure used elsewhere in the library).

Reference: [https://github.com/firebase/FirebaseUI-Android/blob/master/auth/build.gradle.kts](https://github.com/firebase/FirebaseUI-Android/blob/master/auth/build.gradle.kts)

**Comment by samtstern** [comment]
@curiousily thanks for the feedback.  When you ship your app with ProGuard, are you seeing a lot of these classes still hanging around?  I realize that we may be bloating your debug builds but hopefully that's not a blocking issue.

**Comment by ArtbyRere** [comment]
K can you help

**Comment by SUPERCILEX** [comment]
Ok, so only the Twitter dep uses retrofit. If you're not using Twitter, then it is an unnecessary dependency. @samtstern What do you think of making it `compileOnly`? Would that count as a breaking change?

@curiousily BTW, you can always exclude it in the meantime:
```kt
implementation("com.firebaseui:firebase-ui-auth:4.1.0") {
    exclude(module = "retrofit") // In Groovy, I think this would be `exclude module: 'retrofit'
    exclude(module = "converter-gson")
}
```

**Comment by samtstern** [comment]
@SUPERCILEX I think we probably need to change how GitHub works due to the other bug about `CHANGE-ME` showing up in system settings when it's not configured.

I think in theory it's a breaking change but in practice I don't think we should update the major version number for this when I can't see anyone getting "accidentally" broken by it.

**Comment by SUPERCILEX** [comment]
Yeah, I'm thinking we need to pull out a sub-module with the manifest entry and those deps. 👍

**Comment by curiousily** [comment]
Hey guys,

Yes, one can remove the deps using ProGuard or something similar, but that ain't a real solution, as you've probably guessed.

@SUPERCILEX even though we don't use Retrofit (we're hardcore - using OkHttp 😃) I tend to agree with you about it. However, GSON is definitely not the only reasonable JSON lib option 🙂. That said, I will use the proposed workaround for now (exclude the deps).

Thanks for the amazing work you guys are doing with this lib!

**Comment by SUPERCILEX** [comment]
Yeah, don't worry, we are going to fix it. 👍

PS: retrofit actually uses okhttp under the hood.

**Comment by samtstern** [comment]
This was fixed and released in version `4.2.0`
